ext/vulnsrc/suse/suse.go

```diff
@@ -0,0 +1,473 @@
+			log.WithError(err).WithField("ovalFile", ovalFile).Warning("Ignoring OVAL file")
+		}
+
+		if timestamp > latestOval {
+			ovalFiles = append(ovalFiles, ovalFile)
+		}
+	}
+
+	for _, oval := range ovalFiles {
+		// Download the oval XML file.
+		r, err := http.Get(oval)
+		if err != nil {
+			log.WithError(err).Error("could not download", u.Name, "update list")
+			return resp, commonerr.ErrCouldNotDownload
+		}
+
+		match := u.FileRegexp.FindStringSubmatch(oval)
+		if len(match) != 2 {
+			log.Error("Skipping ", oval, "because it's not possible to extract osVersion")
+			continue
+		}
+		osVersion := match[1]
+
+		// Parse the XML.
+		vs, generationTime, err := parseOval(r.Body, u.NamespaceName, osVersion)
+		if err != nil {
+			return resp, err
+		}
+		generationTimes = append(generationTimes, generationTime)
+
+		// Collect vulnerabilities.
+		for _, v := range vs {
+			resp.Vulnerabilities = append(resp.Vulnerabilities, v)
+		}
+	}
+
+	// Set the flag if we found anything.
+	if len(generationTimes) > 0 {
+		resp.FlagName = u.UpdaterFlag
+		resp.FlagValue = strconv.FormatInt(latest(generationTimes), 10)
+	} else {
+		log.WithField("package", u.Name).Debug("no update")
+	}
+
+	return resp, nil
+}
+
+// Get the latest modification time of a remote file
+// expressed as unix time
+func getLatestModifiedTime(url string) (int64, error) {
+	resp, err := http.Head(url)
+	if err != nil {
+		return 0, err
+	}
+
+	last_modified := resp.Header.Get("Last-Modified")
+	if len(last_modified) == 0 {
+		return 0, fmt.Errorf("last modified header missing")
+	}
+
+	// "Thu, 30 Nov 2017 03:07:57 GMT
+	layout := "Mon, 2 Jan 2006 15:04:05 MST"
+	timestamp, err := time.Parse(layout, last_modified)
+	if err != nil {
+		return 0, err
+	}
+
+	return timestamp.Unix(), nil
+}
+
+func latest(values []int64) (ret int64) {
+	for _, element := range values {
+		if element > ret {
+			ret = element
+		}
+	}
+	return
+}
+
+func (u *updater) Clean() {}
+
+func parseOval(ovalReader io.Reader, osFlavor, osVersion string) (vulnerabilities []database.VulnerabilityWithAffected, generationTime int64, err error) {
+	// Decode the XML.
+	var ov oval
+	err = xml.NewDecoder(ovalReader).Decode(&ov)
+	if err != nil {
+		log.WithError(err).Error("could not decode XML")
+		err = commonerr.ErrCouldNotParse
+		return
+	}
+
+	// timestamp format 2017-10-23T04:07:14
+	layout := "2006-1-2T15:04:05"
+	timestamp, err := time.Parse(layout, ov.Timestamp)
+	if err != nil {
+		return
+	}
+	generationTime = timestamp.Unix()
+
+	// Iterate over the definitions and collect any vulnerabilities that affect
+	// at least one package.
+	for _, definition := range ov.Definitions {
+		pkgs := toFeatureVersions(definition.Criteria, osFlavor, osVersion)
+		if len(pkgs) > 0 {
+			vulnerability := database.VulnerabilityWithAffected{
+				Vulnerability: database.Vulnerability{
+					Name: name(definition),
+					Link: link(definition),
+					//TODO: handle that once openSUSE/SLE OVAL files have severity info
```

Comment:
pull this out into a function `severity(definition)` that doesn't do anything but return UnknownSeverity, and has this comment.